lib/applicationModel.js

```diff
@@ -0,0 +1,135 @@
+    this.setupModelClassMap();
+    const modelClass = this.modelClassMap[schemaName];
+    debugApplicationModel(`returning modelClass for  caller ${this.caller} ${schemaName}`);
+    debugApplicationModel(modelClass);
+    return modelClass;
+  };
+
+  getSchema(schemaName) {
+    //debugApplicationModel(ApplicationModel.asyncapi.allSchemas());
+    const schema = ApplicationModel.asyncapi.components().schema(schemaName);
+    //debugApplicationModel(`getSchema for ${schemaName}`);
+    //debugApplicationModel(schema);
+    return schema;
+  }
+
+  setupSuperClassMap() {
+    if (!this.superClassMap) {
+      this.superClassMap = new Map();
+      this.anonymousSchemaToSubClassMap = new Map();
+      debugApplicationModel('-------- SCHEMAS -------------');
+      debugApplicationModel(ApplicationModel.asyncapi.allSchemas());
+      ApplicationModel.asyncapi.allSchemas().forEach((schema, schemaName) => {
+        debugApplicationModel(`${schemaName}:`);
+        debugApplicationModel(schema);
+        const allOf = schema.allOf();
+        if (allOf) {
+          let anonymousSchema;
+          let namedSchema;
+          allOf.forEach(innerSchema => {
+            debugApplicationModel('=== allOf inner schema: ===');
+            debugApplicationModel(innerSchema);
+            debugApplicationModel('===========================');
+            const name = innerSchema._json['x-parser-schema-id'];
+            if (this.isAnonymousSchema(name)) {
+              anonymousSchema = name;
+            } else {
+              namedSchema = name;
+            }
+          });
+          if (!anonymousSchema || !namedSchema) {
+            console.log('Warning: Unable to find both an anonymous and a named schema in an allOf schema.');
+            console.log(schema);
+          } else {
+            this.superClassMap[anonymousSchema] = namedSchema;
+            this.anonymousSchemaToSubClassMap[anonymousSchema] = schemaName;
+          }
+        }
+      });
+      debugApplicationModel('-----------------------------');
+      debugApplicationModel('superclassMap:');
+      debugApplicationModel(this.superClassMap);
+      debugApplicationModel('anonymousSchemaToSubClassMap:');
+      debugApplicationModel(this.anonymousSchemaToSubClassMap);
+    }
+  }
+
+  setupModelClassMap() {
+    if (!this.modelClassMap) {
+      this.modelClassMap = new Map();
+      ApplicationModel.asyncapi.allSchemas().forEach((schema, schemaName) => {
+        debugApplicationModel(`setupModelClassMap ${schemaName} type ${schema.type()}`);
+        const allOf = schema.allOf();
+        debugApplicationModel('allOf:');
+        debugApplicationModel(allOf);
+        if (allOf) {
+          allOf.forEach(innerSchema => {
+            const name = innerSchema._json['x-parser-schema-id'];
+            if (this.isAnonymousSchema(name) && innerSchema.type() === 'object') {
+              this.addSchemaToMap(innerSchema, schemaName);
+            }
```

Comment:
I'm worried about schema.type() being 'object' but not the kind of object we expect. I'd assume the parser does something to make sure it's the right thing, though. typeof null === 'object', typeof [] === 'object', etc...

Reply:
That's a good point. So far I haven't seen any problems with this, but I'll review the parser-js code to make sure.

Reply:
Oh this type is a json schema type, so that could include string, number, integer, object, array, boolean, and null. If the schema type is an array, for example, would we still want a class for it? I think it would be different. So leaving it to do its thing with just 'object' is good enough. Dont want to scope creep.